homeassistant/components/nina/binary_sensor.py

```diff
@@ -0,0 +1,125 @@
+    DEVICE_CLASS_SAFETY,
+    BinarySensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from . import NINADataUpdateCoordinator
+from .const import (
+    ATTR_EXPIRES,
+    ATTR_HEADLINE,
+    ATTR_ID,
+    ATTR_SENT,
+    ATTR_START,
+    CONF_FILTER_CORONA,
+    CONF_MESSAGE_SLOTS,
+    CONF_REGIONS,
+    CORONA_FILTER,
+    DOMAIN,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up entries."""
+
+    coordinator: NINADataUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    filter_corona: bool = config_entry.data[CONF_FILTER_CORONA]
+    regions: dict[str, str] = config_entry.data[CONF_REGIONS]
+    message_slots: int = config_entry.data[CONF_MESSAGE_SLOTS]
+
+    entities: list[NINAMessage] = []
+
+    for ent in coordinator.data:
+        for i in range(0, message_slots):
+            entities.append(
+                NINAMessage(coordinator, ent, regions[ent], i + 1, filter_corona)
+            )
+
+    async_add_entities(entities)
+
+
+class NINAMessage(CoordinatorEntity, BinarySensorEntity):
+    """Representation of an NINA warning."""
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        region: str,
+        regionName: str,
+        slotID: int,
+        filter_corona: bool,
+    ) -> None:
+        """Initialize."""
+        super().__init__(coordinator)
+
+        self._region: str = region
+        self._region_name: str = regionName
+        self._slot_id: int = slotID
+        self._warning_index: int = slotID - 1
+
+        self._filter_corona: bool = filter_corona
+
+        self._coordinator: DataUpdateCoordinator = coordinator
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        return f"Warnung: {self._region_name} {self._slot_id}"
+
+    @property
+    def is_on(self) -> bool:
+        """Return the state of the sensor."""
+        if not len(self._coordinator.data[self._region]) > self._warning_index:
+            return False
+
+        data: dict[str, Any] = self._coordinator.data[self._region][self._warning_index]
+
+        return not (data[CORONA_FILTER] and self._filter_corona)
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
+        """Return extra attributes of the sensor."""
+        if (
+            not len(self._coordinator.data[self._region]) > self._warning_index
+        ) or not self.is_on:
+            return {
+                ATTR_HEADLINE: "",
+                ATTR_ID: "",
+                ATTR_SENT: "",
+                ATTR_START: "",
+                ATTR_EXPIRES: "",
+            }
```

Comment:
```suggestion
            return None
```

There is no need to set empty state attributes